File: 2022/lua/1/main.lua
local function lines_from(file)
  local lines = {}
  for line in io.lines(file) do
    lines[#lines+1] = line
  end
  return lines
end

local function get_totals_from_lines(lines)
  local totals = {}
  local total = 0
  for _, line in pairs(lines) do
    if line == "" then
      totals[#totals+1] = total
      total = 0
    else
      total = total + tonumber(line)
    end
  end
  totals[#totals+1] = total
  return totals
end

local lines = lines_from("input.txt")
local totals = get_totals_from_lines(lines)
table.sort(totals)
print("One: " .. totals[#totals])
print("Two: " .. (totals[#totals] + totals[#totals-1] + totals[#totals-2]))

File: 2022/golang/1/main.go
package main

import (
	"fmt"
	"io/ioutil"
	"sort"
	"strconv"
	"strings"
)

var elfs []Elf

type Elf struct {
	Calories  []int
	TotalCals int
}

func (e *Elf) CalculateTotalCalories() int {
	total := 0
	for _, cal := range e.Calories {
		total += cal
	}
	e.TotalCals = total
	return total
}

func main() {
	prepareInput("input.txt")
	sortElfs()
	one()
	two()
}

func prepareInput(file string) {
	input, _ := ioutil.ReadFile(file)
	lines := strings.Split(strings.TrimSpace(string(input)), "\n\n")
	for _, line := range lines {
		var elf Elf
		for _, cal := range strings.Split(strings.TrimSpace(line), "\n") {
			calInt, _ := strconv.Atoi(cal)
			elf.Calories = append(elf.Calories, calInt)
		}
		elf.CalculateTotalCalories()
		elfs = append(elfs, elf)
	}
}

func sortElfs() {
	sort.Slice(elfs, func(i, j int) bool {
		return elfs[i].TotalCals > elfs[j].TotalCals
	})
}

func one() {
	fmt.Printf("1: %d\n", elfs[0].TotalCals)
}

func two() {
	total := elfs[0].TotalCals + elfs[1].TotalCals + elfs[2].TotalCals
	fmt.Printf("2: %d\n", total)
}
